### PR TITLE
Feature/tag editor backend

### PR DESCRIPTION
**Pull request by garyluu**
This mostly finishes the tag editor for tools.
- Grabs test parameter files using the web service
- deletes and/or add test parameter files
- update the tag
- Implements basic form validation

Current issues:
- Form validation does not check modified test.json file paths, just new ones added
- Form validation combines both cwl and wdl test parameter file validation errors into one
- Both cwl and wdl add parameter file buttons are disabled when either of them are invalid

### CODE REVIEW

File: src/app/container/paramfiles/paramfiles.service.ts [hunk 1]
```diff
@@ -32,6 +34,28 @@ export class ParamfilesService {
     return this.httpService.getResponse(testParamFilesUrl);
   }
 
+  putFiles(containerId: number, testParameterFiles: Array<string>, tagName: string, descriptorType: string) {
+    const url = `${ Dockstore.API_URI }/containers/${ containerId }/testParameterFiles`;
+    const myParams = new URLSearchParams();
+    for (let i = 0; i < testParameterFiles.length; i++) {
```
Comment (denis-yuen) [inline src/app/container/paramfiles/paramfiles.service.ts]: does ng handle arrow functions properly?

File: src/app/shared/validationMessages.model.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      'maxlength': 'Source Code Repository Path is too long. (Max 128 characters.)',
+      'pattern': 'The namespace and name of the Git repository, separated by a \'/\'. ' +
+      'Currently, only GitHub, Bitbucket and GitLab are supported third-party platforms.'
+    },
+    'imagePath': {
+      'required': 'This field cannot be empty.',
+      'minlength': 'Image Path is too short. (Min. 3 characters.)',
+      'maxlength': 'Image Path is too long. (Max 128 characters.)',
+      'pattern': 'The namespace and name of the image repository, separated by a \'/\'. ' +
+      'Use \'_\' for an empty namespace. Currently, only Quay.io and Docker Hub are supported third-party platforms.'
```
Comment (denis-yuen) [inline src/app/shared/validationMessages.model.ts]: Make an issue to update this

Comment (garyluu) [inline src/app/shared/validationMessages.model.ts]: Created an issue

**Comment by garyluu** [comment]
Ready for re-review

**Pull request merged**